src/java.base/share/classes/sun/net/www/protocol/http/AuthenticationInfo.java

```diff
@@ -218,26 +213,20 @@ private static void requestCompleted (String key) {
     /** The shortest path from the URL we authenticated against. */
     String path;
 
-    /**
-     * A key identifying the authenticator from which the credentials
-     * were obtained.
-     * {@link AuthenticatorKeys#DEFAULT} identifies the {@linkplain
-     * java.net.Authenticator#setDefault(java.net.Authenticator) default}
-     * authenticator.
-     */
-     String authenticatorKey;
+    /** A cache to obtain credentials from if the default cache not to be used */
+    AuthCacheImpl authcache;
 
     /** Use this constructor only for proxy entries */
     public AuthenticationInfo(char type, AuthScheme authScheme, String host,
-                              int port, String realm, String authenticatorKey) {
+                              int port, String realm, AuthCacheImpl acache) {
         this.type = type;
         this.authScheme = authScheme;
         this.protocol = "";
         this.host = host.toLowerCase();
         this.port = port;
         this.realm = realm;
         this.path = null;
-        this.authenticatorKey = Objects.requireNonNull(authenticatorKey);
+        this.authcache = acache == null ? defCache : acache;
```

Comment:
Do we really need that field? Now that AuthValue instances are held in separate caches, there should be no need to have a field refering to the authenticator/cache in the auth value.
This let me thinks that some more refactoring could further simplify everything.

Reply:
Possibly - but there seems to be a lot of churn caused by having to pass an AuthCacheImpl instance to the AuthCacheValue now. It makes it a bit difficult to reason about. I just wonder if *not* passing that value would not simplify the fix considerably. I'll try to import your changes and see what I can find.

Reply:
Okay, I'll take a look and see if it can be simplified.

Reply:
I've just pushed an update to the implementation which removes the reference to the cache from the `AuthCacheValue`.